rust-create-cascade/src/main.rs

```diff
@@ -227,20 +233,29 @@ impl From<RevokedSerialAndReasonIterator> for HashMap<Serial, Reason> {
 
 struct KnownSerialIterator {
     lines: Lines<BufReader<File>>,
+    date: u64,
 }
 
 impl KnownSerialIterator {
     fn new(path: &Path) -> Self {
         Self {
             lines: BufReader::new(File::open(path).unwrap()).lines(),
+            date: 0,
         }
     }
 }
 
 impl Iterator for KnownSerialIterator {
-    type Item = String;
+    type Item = (u64, String);
     fn next(&mut self) -> Option<Self::Item> {
-        self.lines.next().transpose().expect("io error")
+        while let Some(line) = self.lines.next().transpose().expect("io error") {
+            if let Some(timestamp) = line.strip_prefix("@") {
+                self.date = u64::from_str_radix(timestamp, 16).expect("malformed date");
+                continue;
+            }
+            return Some((self.date, line));
```

Comment:
There should always be a timestamp first, so we should always have a value for `date`, right? Maybe this should be an `Option` and we assert that it's `Some`?

Reply:
I have old data sets that I use for testing, so I wanted this to work even if there were no "@timestamp" lines.

Reply:
Ok!